### PR TITLE
mgmt, storage enabling blob versioning

### PR DESCRIPTION
**Pull request by XiaofeiCao**
# Description

Fix https://github.com/Azure/azure-sdk-for-java/issues/29105

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/resourcemanager/azure-resourcemanager-storage/src/main/java/com/azure/resourcemanager/storage/models/BlobServiceProperties.java [hunk 1]
```diff
@@ -208,5 +222,25 @@ interface WithDeleteRetentionPolicy {
...
+             */
+            Update withBlobVersioningEnabled();
+
+            /**
+             * Disables blob versioning.
+             * <p>After versioning is disabled, modifying the current version creates a blob that is not a version.
+             * All subsequent updates to the blob will overwrite its data without saving the previous state.
+             * All existing versions persist as previous versions.</p>
+             * <p>You can read or delete versions using the version ID after versioning is disabled.
+             * You can also list a blob's versions after versioning is disabled.</p>
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-storage/src/main/java/com/azure/resourcemanager/storage/models/BlobServiceProperties.java:240]: some of the sentence seems not well formed. e,g. `modifying the current version creates a blob that is not a version` (blob is certainly not a version...)

Comment (XiaofeiCao) [reply sdk/resourcemanager/azure-resourcemanager-storage/src/main/java/com/azure/resourcemanager/storage/models/BlobServiceProperties.java:240]: OK. Didn't realize that when ctrl+c ...

File: sdk/resourcemanager/azure-resourcemanager-storage/CHANGELOG.md [hunk 2]
```diff
@@ -4,9 +4,7 @@
 
 ### Features Added
 
-### Breaking Changes
-
-### Bugs Fixed
+- Supported toggling blob versioning in `BlobServiceProperties`.
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-storage/CHANGELOG.md:7]: Delete other entries (e..g breaking changes etc.).

Or remember to manually modify the changelog in the PR preparing release (the script won't erase them).

Else release pipeline will fail on it (or maybe CI can catch it before release).

**Review by weidongxu-microsoft — APPROVED** [review]

File: sdk/resourcemanager/azure-resourcemanager-storage/CHANGELOG.md (hunk 2 above)
Comment (XiaofeiCao) [reply sdk/resourcemanager/azure-resourcemanager-storage/CHANGELOG.md:7]: OK, thanks!

**Pull request merged**